crate/pkcs11/provider/src/backend.rs

```diff
@@ -161,23 +167,119 @@ impl Backend for CliBackend {
         Ok(result)
     }
 
+    fn find_all_symmetric_keys(&self) -> MResult<Vec<Arc<dyn SymmetricKey>>> {
+        trace!("find_all_symmetric_keys");
+        let mut symmetric_keys = vec![];
+        for id in locate_kms_objects(&self.kms_rest_client, &[])? {
+            let attributes = get_kms_object_attributes(&self.kms_rest_client, &id)?;
+            let key_size = attributes.cryptographic_length.ok_or(MError::Cryptography(
+                "find_all_symmetric_keys: missing key size".to_owned(),
+            ))?;
+            let sk: Arc<dyn SymmetricKey> = Arc::new(Pkcs11SymmetricKey::new(
+                id,
+                key_algorithm_from_attributes(&attributes)?,
+                key_size,
+            ));
+            symmetric_keys.push(sk);
+        }
+
+        Ok(symmetric_keys)
+    }
+
+    fn find_all_keys(&self) -> MResult<Vec<Arc<Object>>> {
+        trace!("find_all_keys");
+        let kms_ids = locate_kms_objects(&self.kms_rest_client, &[])?;
+        let mut objects = Vec::with_capacity(kms_ids.len());
+        for id in kms_ids {
+            let attributes = get_kms_object_attributes(&self.kms_rest_client, &id)?;
+            let key_size = if let Some(key_size) = attributes.cryptographic_length {
+                key_size
+            } else {
+                warn!("find_all_keys: missing key size, skipping {id}");
+                continue;
+            };
+
+            let key_algorithm = key_algorithm_from_attributes(&attributes)?;
+            let object =
+                match attributes.object_type {
```

Comment:
I would use a `if let Some` as you did for `key_size` since it would highlight the fact that they are both handled the same way. Plus a match on an option is not really idiomatic